ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1329,6 +1331,9 @@ private void filterDeleteEntries(List<MessageInfo> messageEntries) {
   void close() throws StoreException {
     long startTimeInMs = time.milliseconds();
     try {
+      if (persistorTask != null) {
+        persistorTask.cancel(false);
+      }
       persistor.write();
```

Comment:
orthogonal to this change, I think there is some value in the persistor keeping track of the offset before flush across `write()` calls so it can avoid unnecessary flushes (may be useful here during shutdown and with RO partitions).